### PR TITLE
rr and valgrind additions

### PR DESCRIPTION
**Pull request by d-w-moore**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -176,3 +183,47 @@ Start a debugger container.
...
+      - install an iRODS server with debug symbols compiled in.
+         (*Can insert **rodsLog(LOG_NOTICE,"...%s..",arg,...)** calls to help ascertaining relevant PID later)*
+      - as iRODS, do: `~/irodsctl stop`
+      - then, also as iRODS: `cd /usr/sbin ; rr record -w /usr/sbin/irodsServer`
+      - (as any user) perform the operations requiring troubleshooting
+      - again as iRODS, do: `~/irodsctl stop`
+      then:
+      ```
+      rr ps
+      rr replay -p <PID> # or if  forked without exec, then "-f <PID>"
```
Comment (trel) [inline README.md:208]: extra space between 'if  forked'

File: README.md [hunk 2]
```diff
@@ -150,6 +150,13 @@ $ ./run_debugger.sh -d .. -V volumes.include.sh --debug
 
 ## Debugging
 
+The system calls necessary for `gdb` and `rr` to function properly on the Linux host necessitate:
+   - `/proc/sys/kernel/yama/ptrace_scope` no higher than 0
+   - `/proc/sys/kernel/perf_event_paranoid` no higher than 1
+   
+These can be changed by editing the config files under `/etc/sysctl.*` and reloaded via `sysctl --system`
```
Comment (trel) [inline README.md:157]: need period at the end of sentence.

File: README.md [hunk 3]
```diff
@@ -150,6 +150,13 @@ $ ./run_debugger.sh -d .. -V volumes.include.sh --debug
...
+(Doing so on the host machine should be effective for all containerized `gdb`/`rr` runs as well.)
```
Comment (trel) [inline README.md:158]: lose the parentheses, and change to 'Doing this on'

File: README.md [hunk 4]
```diff
@@ -176,3 +183,47 @@ Start a debugger container.
 
     run the client to invoke the API
 
+### RR
+
+   - GDB work-a-like
```
Comment (trel) [inline README.md:188]: please use more words here

File: README.md [hunk 5]
```diff
@@ -176,3 +183,47 @@ Start a debugger container.
...
+      - again as iRODS, do: `~/irodsctl stop`
+      then:
+      ```
+      rr ps
+      rr replay -p <PID> # or if  forked without exec, then "-f <PID>"
+      ```
+
+### Valgrind
+
+```
```
Comment (trel) [inline README.md:213]: let's split the comments from the code itself

**Review by trel — COMMENTED** [review]
a couple changes - and we can iterate later.

happy with two commits or one.   no worries.

File: README.md (hunk 1 above)
Comment (d-w-moore) [reply README.md]: These changes are done; guess we'll squash everything once we're satisfied .

**Comment by trel** [comment]
i like it - please squash, remove 'draft'... and i'll merge it

**Comment by d-w-moore** [comment]
Done!

**Pull request merged**